src/renderers/dom/shared/devtools/ReactDOMDuplicateSelectValuesDevtool.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactDOMDuplicateSelectValuesDevtool
+ */
+
+'use strict';
+
+var ReactComponentTreeDevtool = require('ReactComponentTreeDevtool');
+var warning = require('warning');
+
+var didWarnDupeSelectValues = false;
+
+function handleElement(debugID, element) {
+  
+  if (didWarnDupeSelectValues) {
+    return;
+  }
+  
+  if (element == null) {
+    return;
+  }
+
+  if (element.type !== 'select') {
+    return;
+  }
+
+  if ((element.props == null) || (element.props.children == null)) {
+    return;
+  }
+
+  // Uncontrolled select elements can have duplicate values.
+  if (element.props.hasOwnProperty('defaultValue')) {
+    return;
+  }
+
+  let values = {};
+  let options = element.props.children;
+
+  //  If options is not iterable make it an array.
+  if (typeof options[Symbol.iterator] !== 'function') {
+    options = [options];
+  }
+
+  for (const option of options) {
+    if (option.type === 'optGroup') {
```

Comment:
Do you mean the `props.children` for a `select` element can be something besides `option` or `optGroup`? I'm not sure how that would work. Can you point me towards some documentation  explaining it? Also, thanks for the feedback. Researching these issues you and others have pointed out has taught me quite a bit about how React works.